libraries/AP_Camera/AP_Camera.cpp

```diff
@@ -379,26 +392,17 @@ void AP_Camera::update_trigger()
 {
     trigger_pic_cleanup();
     
-    bool triggered;
-    uint64_t timestamp;
-    
-    {
-        WITH_SEMAPHORE(_feedback_sem);
-        triggered = _camera_triggered;
-        timestamp = _feedback_timestamp_us;
-        _camera_triggered = false;
-    }
-    
-    if (triggered) {
-        _feedback_events++;
+    if (_camera_trigger_logged != _camera_trigger_count) {
+        uint32_t timestamp32 = _feedback_timestamp_us;
+        _camera_trigger_logged++;
```

Comment:
Should we make it equal to `_camera_trigger_count`? Example where current code would be problematic:

- in one loop, `_camera_trigger_count` is increased twice before `update_trigger()` is called and logging happens
- logging will continue to happen as numbers won't be equal
- in another loop, camera isn't actually triggered but logging will still happen with previous timestamp

This might be very rare to impossibly happening, but the fix is also easy.

Reply:
fixed, thanks